packages/mobile/src/localCurrency/sagas.ts

```diff
@@ -0,0 +1,74 @@
+import gql from 'graphql-tag'
+import { call, delay, put, spawn } from 'redux-saga/effects'
+import { apolloClient } from 'src/apollo'
+import { waitForRehydrate } from 'src/app/saga'
+import { LOCAL_CURRENCY_SYMBOL } from 'src/config'
+import {
+  fetchCurrentRate,
+  fetchCurrentRateFailure,
+  fetchCurrentRateSuccess,
+} from 'src/localCurrency/actions'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'localCurrency/saga'
+
+const POLLING_DELAY = 3600 * 1000 // 1 hour
+
+// Delay before retrying when an error is raised
+const RETRY_DELAY = 10 * 1000 // 10 secs
+
+async function fetchExchangeRate(symbol: string): Promise<number> {
+  const response = await apolloClient.query({
+    query: gql`
+    {    
+      currencyConversion(currencyCode: "${symbol}") {
+        rate
+      }
+    }
+  `,
+    fetchPolicy: 'network-only',
+    errorPolicy: 'all',
+  })
+
+  const { data } = response
+  if (typeof data !== 'object') {
+    throw new Error(`Invalid response data ${data}`)
+  }
+
+  const { currencyConversion } = data
+  if (typeof currencyConversion !== 'object') {
+    throw new Error(`Invalid response data ${data}`)
+  }
+
+  const { rate } = currencyConversion
+  if (typeof rate !== 'number') {
+    throw new Error(`Invalid response data ${data}`)
+  }
+
+  return rate
+}
+
+function* updateLocalCurrencyRate() {
+  if (!LOCAL_CURRENCY_SYMBOL) {
+    return
+  }
+
+  yield call(waitForRehydrate)
+
+  while (true) {
+    try {
+      yield put(fetchCurrentRate())
```

Comment:
The naming here confuses me a bit, I see this and I expect it to trigger a watcher that will do the fetching, but actually we do the fetching right here

Reply:
Agreed. I'll follow up with that in another PR.